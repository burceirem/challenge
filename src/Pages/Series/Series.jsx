import React ,{useState ,useEffect}from "react";
import PropTypes from 'prop-types';
import Header from "../../Components/Header/Header";
import Footer from "../../Components/Footer/Footer";
import Content from "../../Components/Content/Content";
import { Dropdown ,Navbar} from 'react-bootstrap';
import Searchbar from '../../Components/Searchbar/Searchbar';
import '../MoviesAndSeries.css';

const Series = ({seriesData}) => {
 
    const [sort, setSort] = useState("");
    const [searchData, setSearchData] = useState("");
    const [searchTerm, setSearchTerm] = useState("");
   
    
 const myCallback = (searchTerm , sort) => {
  setSearchTerm(searchTerm);
  setSort(sort);
 
};


    useEffect(() => {
     let results= seriesData.filter(item =>
      item.title.toLowerCase().includes(searchTerm)
      );
      setSearchData(results);
    }, [searchTerm]);

    const getSortedData = (sortKey) => { 
     if(sortKey === 'yearAsc') {
        return [].slice.call(seriesData).sort(sortYearAsc);
     }
     if(sortKey === 'yearDesc') {
        return [].slice.call(seriesData).sort(sortYearDesc);
      }
      if(sortKey === 'titleDesc'){
        return [].slice.call(seriesData).sort(sortTitleDesc);
      }
      if(sortKey ==='search'){
        return searchData
      }

     return [].slice.call(seriesData).sort(sortTitleAsc);
    };

    const handleSelect = (sortKey) => {
      setSort(sortKey);
 
    };
    const sortTitleAsc = (a, b ) => {
      const titleA = a.title.toUpperCase();
      const titleB = b.title.toUpperCase();
      let comparison = 0;

      if (titleA > titleB) {
        comparison = 1;
      } else if (titleA < titleB) {
        comparison = -1;
      }
      return comparison;
    };

    const sortTitleDesc = (a, b ) => {
      const titleA = a.title.toUpperCase();
      const titleB = b.title.toUpperCase();
      let comparison = 0;

      if (titleA > titleB) {
        comparison = -1;
      } else if (titleA < titleB) {
        comparison = 1;
      }
      return comparison;
    };
    const sortYearAsc = (a,b) => {
        return  a.releaseYear-b.releaseYear;
     
    };

    const sortYearDesc = (a,b) => {
      return  b.releaseYear-a.releaseYear;
   
  };


    
    return(
    <React.Fragment>
        <Header/>
        <Navbar   data-testid="navbar" className="navbar"> 
        <Searchbar  callbackFromParent={myCallback}/>
        <Dropdown onSelect={handleSelect} className="dropdown">
        <Dropdown.Toggle variant="success" id="dropdown-basic">
          Sort
        </Dropdown.Toggle>
        <Dropdown.Menu>
          <Dropdown.Item eventKey="titleAsc">Title Asc</Dropdown.Item>
          <Dropdown.Item eventKey="titleDesc">Title Desc</Dropdown.Item>
          <Dropdown.Item eventKey="yearAsc">Release Year Asc</Dropdown.Item> 
          <Dropdown.Item eventKey="yearDesc">Release Year Desc</Dropdown.Item> 
        </Dropdown.Menu>
      </Dropdown>
      </Navbar>
      <div  className="content-container"> 
          {getSortedData(sort).map (  v =>  <Content key={v.title} title = {v.title} url ={v.images.PosterArt.url} /> )} 
      </div>
      <Footer/>
    </React.Fragment>);
};

Series.propTypes = {
  seriesData: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      description: PropTypes.string,
      programType: PropTypes.string,
      images: PropTypes.shape({}),
      releaseYear:PropTypes.number
    }),
  ).isRequired,
};

export default Series;