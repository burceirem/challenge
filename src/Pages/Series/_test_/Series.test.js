import React from "react";
import { render ,fireEvent  } from "@testing-library/react";
import renderer from "react-test-renderer";
import Series from "../Series";


const data = [ 
                {title: "Wolf Creek", 
                description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, do consequat.",
                programType: "series",
                images: {
                  "PosterArt": {
                    "url": "https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg",
                    "width": 1000,
                    "height": 1500
                  }
                },
                releaseYear: 2016
                }];
  

  it("Check click event of sort button", () => {
    const { getByText} = render(<Series  seriesData={data}/>);
    fireEvent.click(getByText("Sort"));
  });

  it("Check searchbar placeholder text", () => {
    const { getByPlaceholderText} = render(<Series  seriesData={data}/>);
    fireEvent.click(getByPlaceholderText("Search"));
  });

  it("matches snapshot",() => {
    const tree = renderer.create(<Series  seriesData={data}/>);
    expect(tree).toMatchSnapshot();
  });

