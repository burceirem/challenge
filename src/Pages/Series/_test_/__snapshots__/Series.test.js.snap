// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="header"
    data-testid="header"
  >
    <nav
      className="navbar navbar-expand navbar-light bg-light"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        Homepage
      </a>
    </nav>
  </div>,
  <nav
    className="navbar navbar navbar-expand navbar-light"
    data-testid="navbar"
  >
    <form
      className="form-inline"
    >
      <input
        className="mr-sm-2 form-control"
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
    </form>
    <div
      className="dropdown dropdown"
      onKeyDown={[Function]}
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-success"
        disabled={false}
        id="dropdown-basic"
        onClick={[Function]}
        type="button"
      >
        Sort
      </button>
    </div>
  </nav>,
  <div
    className="content-container"
  >
    <div
      className="card card"
      data-testid="card"
    >
      <img
        className="card-img-top"
        src="https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg"
      />
      <div
        className="card-body"
      >
        <p
          className="card-text"
        >
           
          Wolf Creek
           
        </p>
      </div>
    </div>
  </div>,
  <div
    className="footer"
    data-testid="footer"
  >
    <p>
      PANKOD CHALLENGE PROJECT 
    </p>
  </div>,
]
`;
